### ISSUE TITLE
[BUG] cannot write @ character on nordic (Finnish, Swedish) keyboard

### ISSUE THREAD

**Issue by villesau** [anthropics/claude-code#8259]
### Preflight Checklist

- [x] I have searched [existing issues](https://github.com/anthropics/claude-code/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug) and this hasn't been reported yet
- [x] This is a single bug report (please file separate reports for different bugs)
- [x] I am using the latest version of Claude Code

### What's Wrong?

Claude code does not write @ sign on nordic keyboards. This is since it uses option as meta key, whereas nordic keyboards require you to press option+2 for writing characters such as @. Claude also does not use zsh keybindings that overrides the behavior: https://gist.github.com/villesau/ebe834fd943f8ef0eb39ef96dcfa5788 

### What Should Happen?

option+2 (writing @ sign) should write @ character.

### Error Messages/Logs

```shell
None
```

### Steps to Reproduce

Use nordic keyboard (finnish, swedish) and try to write @ using option+2. Using nordic keyboard layout might be enough to reproduce. Might require to toggle "Use Option as Meta key" toggled on in the settings. If it is not toggled in the settings, writing @ character works but option + enter does not insert new line so it is similarly handicapped.

### Claude Model

Sonnet (default)

### Is this a regression?

No, this never worked

### Claude Code Version

v1.0.127

### Platform

Anthropic API

### Operating System

macOS

### Terminal/Shell

Terminal.app (macOS)

**Comment by villesau** [comment]
These kind of issues are very related:
- https://github.com/anthropics/claude-code/issues/7958
- https://github.com/anthropics/claude-code/issues/1793
- https://github.com/anthropics/claude-code/issues/1297
- https://github.com/anthropics/claude-code/issues/2751

**Comment by mdelapenya** [comment]
Same happens with the Spanish layout

**Comment by mdelapenya** [comment]
As of today, it's still happening to me using the Spanish layout. I'm forced to switch to US layout to type `@`

**Comment by mdelapenya** [comment]
Hi, the issue is still relevant, could you please re-open it so the team can allocate time for this?

**Comment by fsc-eriker** [comment]
Isn't everything documented in https://github.com/anthropics/claude-code/issues/2751 though?

**Comment by mdelapenya** [comment]
@fsc-eriker I did not know about that issue, but reading through it tells me that I need to switch my terminal or my settings, right?

**Comment by fsc-eriker** [comment]
That is my understanding, yes; but like all of these bugs with multiple partially overlapping duplicates, it's hard to see what exactly is the community's consensus or the company's stance on doing something about the problem(s). It would be helpful if they actively merged duplicates instead of waiting for bugs to auto-close as if they didn't plan to do anything about this ever.

**Comment by marcindulak** [comment]
This issue was closed incorrectly despite recent human comments. This behavior of the bot is reported at https://github.com/anthropics/claude-code/issues/16497. Please upvote that issue, so maybe it gets noticed.
